velox/dwio/common/compression/Compression.h

```diff
@@ -60,20 +67,70 @@ class Decompressor {
   const std::string streamDebugInfo_;
 };
 
+struct CompressionOptions {
+  union Format {
+    struct {
+      int windowBits;
+      int32_t compressionLevel;
+    } zlib;
+
+    struct {
+      int32_t compressionLevel;
+    } zstd;
+  } format;
+
+  uint32_t compressionThreshold;
+};
+
+static CompressionOptions getDwrfOrcCompressionOptions(
+    velox::common::CompressionKind kind,
+    uint32_t compressionThreshold,
+    int32_t zlibCompressionLevel,
+    int32_t zstdCompressionLevel) {
+  CompressionOptions options;
+  options.compressionThreshold = compressionThreshold;
+
+  if (kind == velox::common::CompressionKind_ZLIB) {
+    options.format.zlib.windowBits = Compressor::DWRF_ORC_ZLIB_WINDOW_BITS;
+    options.format.zlib.compressionLevel = zlibCompressionLevel;
+  } else if (kind == velox::common::CompressionKind_ZSTD) {
+    options.format.zstd.compressionLevel = zstdCompressionLevel;
+  }
+  return options;
+}
+
+static CompressionOptions getDwrfOrcDecompressionOptions() {
+  CompressionOptions options;
+  options.format.zlib.windowBits = Compressor::DWRF_ORC_ZLIB_WINDOW_BITS;
+  return options;
+}
+
+static CompressionOptions getParquetDecompressionOptions() {
+  CompressionOptions options;
```

Comment:
```
const CompressionOptions& options;
```

Reply:
can't declare it const reference. needs initialization.